### PR TITLE
Update browserlist

### PR DESCRIPTION
**Pull request by sneridagh**
- [ ] I signed and returned the [Plone Contributor Agreement](https://plone.org/foundation/contributors-agreement), and received and accepted an invitation to join a team in the Plone GitHub organization.
- [ ] I verified there aren't other open [pull requests](https://github.com/plone/volto/pulls) for the same change.
- [ ] I followed the guidelines in [Contributing to Volto](https://6.docs.plone.org/volto/contributing/index.html).
- [ ] I succesfully ran [code linting checks](https://6.docs.plone.org/volto/contributing/linting.html) on my changes locally.
- [ ] I succesfully ran [unit tests](https://6.docs.plone.org/volto/contributing/testing.html) on my changes locally.
- [ ] I succesfully ran [acceptance tests](https://6.docs.plone.org/volto/contributing/acceptance-tests.html) on my changes locally.
- [ ] If needed, I added new tests for my changes.
- [ ] If needed, I added [documentation](https://6.docs.plone.org/volto/contributing/documentation.html#narrative-documentation) for my changes, either in the Storybook or narrative documentation.
- [ ] I included a [change log entry](https://6.docs.plone.org/contributing/index.html#contributing-change-log-label) in my commits.

-----

If your pull request closes an open issue, include the exact text below, immediately followed by the issue number. When your pull request gets merged, then that issue will close automatically.

Closes #

### CODE REVIEW

**Comment by stevepiercy** [comment]
https://www.npmjs.com/package/browserlist says use https://github.com/browserslist/browserslist. browser**s**list. Should we use GitHub directly instead of npmjs.com?

Also https://6.docs.plone.org/volto/contributing/language-features.html#browser-compatibility should be updated, if you change that requirement. There's also another typo.

File: package.json [hunk 1]
```diff
@@ -27,7 +27,8 @@
     "lockhook": "node packages/scripts/lockhook.js",
     "prepare": "husky",
     "husky:uninstall": "husky uninstall",
-    "prereleaser": "node packages/scripts/preleaser.js"
+    "prereleaser": "node packages/scripts/preleaser.js",
+    "update:browserlist": "pnpm --filter @plone/volto add caniuse-lite && pnpm --filter @plone/volto remove caniuse-lite"
```
Comment (davisagli) [inline package.json:31]: @sneridagh Despite removing it in this command, `caniuse-lite` ended up in the lockfile. Is that a mistake?

**Comment by davisagli** [comment]
@stevepiercy We are in fact using browserslist, not browserlist. It is misspelled in the title of this PR and a couple places in https://6.docs.plone.org/volto/contributing/language-features.html#browser-compatibility

Updating the package doesn't change the rules that we are using it to apply.

**Comment by stevepiercy** [comment]
Addressed in https://github.com/plone/volto/commit/528ef97cc26998029e6c67c2baa9e7e45eb5238f as part of https://github.com/plone/volto/pull/6397

**Pull request merged**